extensions/azurecore/src/account-provider/azureAccountProvider.ts

```diff
@@ -3,446 +3,155 @@
 	}
 
-	private isPromptFailed(value: adal.TokenResponse | azdata.PromptFailedResult): value is azdata.PromptFailedResult {
-		return value && (<azdata.PromptFailedResult>value).canceled;
+	private async _clear(accountKey: azdata.AccountKey): Promise<void> {
+		await this.initCompletePromise;
+		await this.getAuthMethod(undefined)?.clearCredentials(accountKey);
 	}
 
-	private async signIn(isAddAccount: boolean): Promise<AzureAccount | azdata.PromptFailedResult> {
-		// 1) Get the user code for this login
-		// 2) Get an access token from the device code
-		// 3) Get the list of tenants
-		// 4) Generate the AzureAccount object and return it
-		let tokenResponse: adal.TokenResponse = null;
-		let result: InProgressAutoOAuth = await this.getDeviceLoginUserCode();
-		this._autoOAuthCancelled = false;
-		this._inProgressAutoOAuth = result;
-		let response: adal.TokenResponse | azdata.PromptFailedResult = await this.getDeviceLoginToken(this._inProgressAutoOAuth, isAddAccount);
-		if (this.isPromptFailed(response)) {
-			return response;
-		}
-		tokenResponse = response;
-		this._autoOAuthCancelled = false;
-		this._inProgressAutoOAuth = null;
-		let tenants: Tenant[] = await this.getTenants(tokenResponse.userId, tokenResponse.userId);
-		// Figure out where we're getting the identity from
-		let identityProvider = tokenResponse.identityProvider;
-		if (identityProvider) {
-			identityProvider = identityProvider.toLowerCase();
-		}
-
-		// Determine if this is a microsoft account
-		let msa = identityProvider && (
-			identityProvider.indexOf('live.com') !== -1 ||
-			identityProvider.indexOf('live-int.com') !== -1 ||
-			identityProvider.indexOf('f8cdef31-a31e-4b4a-93e4-5f571e91255a') !== -1 ||
-			identityProvider.indexOf('ea8a4392-515e-481f-879e-6571ff2a8a36') !== -1);
-
-		// Calculate the display name for the user
-		let displayName = (tokenResponse.givenName && tokenResponse.familyName)
-			? `${tokenResponse.givenName} ${tokenResponse.familyName}`
-			: tokenResponse.userId;
-
-		// Calculate the home tenant display name to use for the contextual display name
-		let contextualDisplayName = msa
-			? localize('microsoftAccountDisplayName', "Microsoft Account")
-			: tenants[0].displayName;
-
-		// Calculate the account type
-		let accountType = msa
-			? AzureAccountProvider.MicrosoftAccountType
-			: AzureAccountProvider.WorkSchoolAccountType;
-
-		return <AzureAccount>{
-			key: {
-				providerId: this._metadata.id,
-				accountId: tokenResponse.userId
-			},
-			name: tokenResponse.userId,
-			displayInfo: {
-				accountType: accountType,
-				userId: tokenResponse.userId,
-				contextualDisplayName: contextualDisplayName,
-				displayName: displayName
-			},
-			properties: {
-				providerSettings: this._metadata,
-				isMsAccount: msa,
-				tenants: tenants
-			},
-			isStale: false
-		};
+	autoOAuthCancelled(): Thenable<void> {
+		this.authMappings.forEach(val => val.autoOAuthCancelled());
+		return undefined;
```

Comment:
I'd rather be safer here and return Promise.resolve(); on the off chance someone is calling .then on autoOAuthCancelled calls.

Reply:
Good idea.